standard/countdown.lua

```diff
@@ -0,0 +1,77 @@
+---
+-- @Liquipedia
+-- wiki=commons
+-- page=Module:Countdown
+--
+-- Please see https://github.com/Liquipedia/Lua-Modules to contribute
+--
+
+local Arguments = require('Module:Arguments')
+local DateExt = require('Module:Date/Ext')
+local Logic = require('Module:Logic')
+local Lua = require('Module:Lua')
+
+local StreamLinks = Lua.import('Module:Links/Stream')
+
+local NOW = os.time()
+
+local Countdown = {}
+
+---@param frame Frame
+---@return string
+function Countdown.create(frame)
+	return Countdown._create(Arguments.getArgs(frame))
+end
+
+---@param args table
+---@return string
+function Countdown._create(args)
+	if Logic.isEmpty(args.date) and not args.timestamp then
+		return ''
+	end
+
+	local wrapper = mw.html.create('span')
+		:addClass('timer-object')
+
+	if Logic.readBool(args.rawcountdown) then
+		wrapper:addClass('timer-object-countdown-only')
+	end
+	if Logic.readBool(args.rawdatetime) then
+		wrapper:addClass('timer-object-datetime-only')
+	end
+
+	-- Timestamp
+	local timestamp = args.timestamp or DateExt.readTimestampOrNil(args.date) or 'error'
+	wrapper:attr('data-timestamp', timestamp)
+
+	local streams
+	local isFinished = Logic.readBool(args.finished)
+		-- the js assumes a match finished if the match is live for 12 hours
+		or (NOW >= timestamp + 43200)
```

Comment:
This is 8 hours in match2 (in general) right?

Also if JS already does this switch, is it needed here as well?

Reply:
js only does the switch from `LIVE` --> Date
the lua part checks if it should append the streams or not
as per https://github.com/Liquipedia/Lua-Modules/pull/4363#discussion_r1652711747 the stream icon building should be moved form the js to lua

Reply:
I would swap the _hiding_ of the streams into the same JS logic that does that switch. Otherwise you'll run into edge cases with inconsistent design. Lua doing the icon building is just fine

Reply:
hmm, so basically if the date condition is met set the child elelment to empty string via js, right?
if so can look into it tomorrow or friday (or next week)